pkg/controller/sync/accessor.go

```diff
@@ -0,0 +1,275 @@
+	}
+	a.templateStore, a.templateController = util.NewResourceInformer(templateClient, targetNamespace, enqueueObj)
+
+	overrideAPIResource := typeConfig.GetOverride()
+	overrideClient, err := util.NewResourceClient(controllerConfig.KubeConfig, &overrideAPIResource)
+	if err != nil {
+		return nil, err
+	}
+	a.overrideStore, a.overrideController = util.NewResourceInformer(overrideClient, targetNamespace, enqueueObj)
+
+	placementAPIResource := typeConfig.GetPlacement()
+	placementClient, err := util.NewResourceClient(controllerConfig.KubeConfig, &placementAPIResource)
+	if err != nil {
+		return nil, err
+	}
+	if typeConfig.GetNamespaced() {
+		namespacePlacementClient, err := util.NewResourceClient(controllerConfig.KubeConfig, namespacePlacementAPIResource)
+		if err != nil {
+			return nil, err
+		}
+		a.placementPlugin = placement.NewNamespacedPlacementPlugin(placementClient, namespacePlacementClient, targetNamespace, enqueueObj)
+	} else {
+		a.placementPlugin = placement.NewResourcePlacementPlugin(placementClient, targetNamespace, enqueueObj)
+	}
+
+	a.versionManager = version.NewVersionManager(
+		fedClient,
+		typeConfig.GetFederatedNamespaced(),
+		typeConfig.GetFederatedKind(),
+		typeConfig.GetTarget().Kind,
+		targetNamespace,
+	)
+
+	// Most types apply the finalizer to the template.
+	finalizerClient := templateClient
+	if a.targetIsNamespace {
+		// For namespaces the finalizer is applied to the placement to
+		// ensure that deletion of namespaces in member clusters only
+		// occurs after a namespace has been configured to be propagated
+		// to those clusters.
+		finalizerClient = placementClient
+	}
+
+	a.deletionHelper = deletionhelper.NewDeletionHelper(
+		func(rawObj pkgruntime.Object) (pkgruntime.Object, error) {
+			obj := rawObj.(*unstructured.Unstructured)
+			return finalizerClient.Resources(obj.GetNamespace()).Update(obj, metav1.UpdateOptions{})
+		},
+		func(obj pkgruntime.Object) string {
+			return util.NewQualifiedName(obj).String()
+		},
+		informer,
+		updater,
+	)
+
+	return a, nil
+}
+
+func (a *resourceAccessor) Run(stopChan <-chan struct{}) {
+	go a.versionManager.Sync(stopChan)
+	go a.templateController.Run(stopChan)
+	go a.overrideController.Run(stopChan)
+	go a.placementPlugin.Run(stopChan)
+}
+
+func (a *resourceAccessor) HasSynced() bool {
+	if !a.versionManager.HasSynced() {
+		glog.V(2).Infof("Version manager not synced")
+		return false
+	}
+	if !a.templateController.HasSynced() {
+		glog.V(2).Infof("Templates not synced")
+		return false
+	}
+	if !a.overrideController.HasSynced() {
+		glog.V(2).Infof("Overrides not synced")
+		return false
+	}
+	if !a.placementPlugin.HasSynced() {
+		glog.V(2).Infof("Placements not synced")
+		return false
+	}
+	return true
+}
+
+func (a *resourceAccessor) FederatedResource(eventSource util.QualifiedName) (FederatedResource, error) {
+	if a.targetIsNamespace && a.isSystemNamespace(eventSource.Name) {
+		glog.V(7).Infof("Ignoring system namespace %q", eventSource.Name)
```

Comment:
question: any reasaon to choose level 7 for this message. Is there any rule in using level of glog.V?

Reply:
The value of logging when a resource is ignored is extremely low, hence the low log level.